File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    progresses = progresses.map((progress, i) =>
        Math.ceil((100 - progress) / speeds[i])
    );

    let answer = [0];
    let day = progresses[0];
    let index = 0;

    for (const progress of progresses) {
        if (progress <= day) {
            answer[index] += 1;
        } else {
            day = progress;
            answer[++index] = 1;
        }
    }

    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math


def solution(progresses, speeds):
    days = [math.ceil((100 - progress)/speed) for progress, speed in zip(progresses, speeds)]
    answer = []
    lead = days[0]
    cnt = 1

    for day in days[1:]:
        if lead >= day:
            cnt += 1
        else:
            answer.append(cnt)
            lead = day
            cnt = 1

    answer.append(cnt)

    return answer